### PR TITLE
fix(macros/JSRef): rewrite and show correct subpages

### PR DESCRIPTION
**Pull request by yin1999**
## Summary

Fixes: #7870

### Problem

When we are visiting sub pages of `Intl`, such as [Intl.Collator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator), [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat), etc.

The sidebar won't show the properties and methods of those `class` type pages.

### Solution

When the current page is under `Intl`, check if the page is under a `class` type sub page. If so, use `Intl.<class-name>`
as the mainObj. And we could find the methods and properties of this.

Also removed `WebAssembly` section as those pages are now using `{{WebAssemblySidebar}}`. ~But we still need to replace all of the old usages in translated-content, see: mdn/translated-content#11430~ (Resolved)

Thoughs: maybe we could use `page-type` metadate to resolve this, but we may need to fetch some parent pages to check if it is a `javascript-class` page. This may have an impact on performance. I just use the groupData the check this.

---

## How did you test this change?

Run `yarn dev` and checked those pages:

- `http://localhost:5042/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator`
- `http://localhost:5042/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/getCanonicalLocales`
- `http://localhost:5042/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl`
- `http://localhost:5042/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat`
- `http://localhost:5042/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array`

### CODE REVIEW

File: kumascript/macros/JSRef.ejs [hunk 1]
```diff
@@ -10,268 +10,247 @@
...
+  Related: mdn.getLocalString(commonl10n, 'Related_pages_wo_group'),
 };
 
-
 // Variables
-var containsTag = page.hasTag;
-var slug = env.slug;
-var output = "";
-// slug is not available in preview mode.
-if (slug) {
```
Comment (yin1999) [inline kumascript/macros/JSRef.ejs:31]: I'm not sure could we remove this check?

File: kumascript/macros/JSRef.ejs [hunk 2]
```diff
@@ -10,268 +10,247 @@
...
-        output += buildSublist(resultConstructors, text['Constructor'], resultOpen);
-    }
-    if (resultProperties.length > 0) {
-        output += await buildSublist(resultProperties, text['Properties'], resultOpen);
-    }
-    if (resultMethods.length > 0) {
-        output += await buildSublist(resultMethods, text['Methods'], resultOpen);
-    }
+  if (resultConstructors.length > 0) {
+    output += await buildSublist(resultConstructors, text.Constructor, resultOpen);
```
Comment (yin1999) [inline kumascript/macros/JSRef.ejs:235]: add `await` to resolve this promise

**Comment by yin1999** [comment]
Hi @NiedziolkaMichal, would you like to have some test on this, or any suggestions?

**Comment by NiedziolkaMichal** [comment]
@yin1999 Thank you for the fix, it works fine. I have added 3 comments about the implementation.

**Comment by yin1999** [comment]
> @yin1999 Thank you for the fix, it works fine. I have added 3 comments about the implementation.

I couldn't see the comments 🥺, still in pending?

File: kumascript/macros/JSRef.ejs [hunk 3]
```diff
@@ -10,268 +10,247 @@
...
-    result += pageBadges;
+  const getTitle = locale === "en-US"
+    ? page => page.title
+    : page => page.translations.find(translation => translation.locale === locale)?.title ?? page.title;
 
-    if (rtlLocales.indexOf(locale) != -1) {
-        result += '</bdi>';
-    }
+  const items = pages.map(async aPage => {
+    const url = aPage.url.replace("en-US", locale);
```
[the remark this answers is not in the document]
Comment (yin1999) [reply kumascript/macros/JSRef.ejs:193]: Now, we just use the same slug for translated content. So we could just replace "en-US" with the current locale

See line 183 and line 189 of the original file.

File: kumascript/macros/JSRef.ejs [hunk 4]
```diff
@@ -10,268 +10,247 @@
...
 
-
 // Variables
-var containsTag = page.hasTag;
-var slug = env.slug;
-var output = "";
-// slug is not available in preview mode.
-if (slug) {
-var locale = env.locale;
-var rtlLocales = ['ar', 'he', 'fa'];
```
[the remark this answers is not in the document]
Comment (yin1999) [reply kumascript/macros/JSRef.ejs:33]: Added back

File: kumascript/macros/JSRef.ejs [hunk 5]
```diff
@@ -10,268 +10,247 @@
...
-var slug_stdlib = `/${env.locale}/docs/Web/JavaScript/Reference/Global_Objects`;
-var mainObj = slug.replace('Web/JavaScript/Reference/Global_Objects/', '').split('/')[0];
+const slug = env.slug;
+let output = "";
+const locale = env.locale;
+const slugStdlib = `/${env.locale}/docs/Web/JavaScript/Reference/Global_Objects`;
+const subSlug = slug.replace('Web/JavaScript/Reference/Global_Objects/', '');
+let subSlugParts = subSlug.split('/');
```
[the remark this answers is not in the document]
Comment (yin1999) [reply kumascript/macros/JSRef.ejs]: renamed

File: kumascript/macros/JSRef.ejs [hunk 6]
```diff
@@ -10,268 +10,238 @@
...
-                title = translation.title;
-            }
-        });
-    }
-
-    const pageBadges = (await page.badges(aPage)).join("");
-
-    if (containsTag(aPage, 'Obsolete')) {
-        result += '<s class="obsoleteElement">';
-    }
```
Comment (yin1999) [inline kumascript/macros/JSRef.ejs:199]: There is no left in mdn/content, just remove this part

**Review by NiedziolkaMichal — APPROVED** [review]

**Review by fiji-flo — APPROVED** [review]
Thank you. This looks great. And thanks for all the modernization!

**Pull request merged**